### PR TITLE
[FEATURE] Provide "estimation histogram" ParameterBuilder output details .

### PR DESCRIPTION
**Pull request by alexsherstinsky**
### Scope
* Introduce `NumericRangeEstimationResult` `dataclass` to contain the estimated value range and the histogram of the estimated statistic's samples.  In the case of the `bootstrap` estimator, all `bootstrap` samples are concatenated for computing the histogram, since this represents a correct approximation to the realization of the underlying distribution.
* The estimation histogram (broken into the default of 10 buckets) is added to the profiler details.  The goal of this new detail is to enable a qualitative confirmation that the distribution of statistics is near-normal as per the bootstrap method's inherent assumption (and increase the number of bootstrap re-samples if it falls short of the near-normality).
* Unit tests for the new histogram capability are added, where the actual and expected histograms are compared using the 2-sample Kolmogorov-Smirnoff statistic, whereby the `p-value` has to be above the commonly accepted value of `0.95` in order to confirm that the null hypothesis cannot be rejected (i.e., the distribution behind the actual histogram is statistically close to that behind the expected histogram).
* Unit tests of histogram computation are provided for both `oneshot` and `bootstrap` estimators; however, the histograms are deleted from the `profiler_details` of the `ExpectationConfiguration` `meta` fields for integration testing (since the `ExpectationConfiguration` objects in corresponding fixtures are too numerous to check individually).
* Clean up type hints, fix typos, and insure that `profiler_details` is the key (not `details`) and is used consistently.

Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

Changes proposed in this pull request:
- JIRA: GREAT-699/GREAT-709
-
-

After submitting your PR, CI checks will run and @cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: great_expectations/rule_based_profiler/helpers/util.py [hunk 1]
```diff
@@ -557,9 +564,14 @@ def compute_bootstrap_quantiles_point_estimate(
             bootstrap_upper_quantile_point_estimate - bootstrap_upper_quantile_bias
         )
 
-    return (
-        lower_quantile_bias_corrected_point_estimate,
-        upper_quantile_bias_corrected_point_estimate,
+    return NumericRangeEstimationResult(
+        estimation_histogram=np.histogram(
+            a=bootstraps.flatten(), bins=NUM_HISTOGRAM_BINS
```
Comment (NathanFarmer) [inline great_expectations/rule_based_profiler/helpers/util.py:569]: This is tricky due to the "custom" nature of our implementation. We should discuss this. The bootstraps are not what is expected to be normal, the computation of the statistic on the bootstraps is.

**Review by NathanFarmer — CHANGES_REQUESTED** [review]
Let's discuss!

File: great_expectations/rule_based_profiler/helpers/util.py (hunk 1 above)
Comment (alexsherstinsky) [reply great_expectations/rule_based_profiler/helpers/util.py:569]: @alexsherstinsky don't our bootstraps consist of the samples of the statistic, which is what we expect to be `Normal`?  And -- yes -- this is particular for our `custom` use.  Is there a way out of this "specificity" for us?..

Comment (NathanFarmer) [reply great_expectations/rule_based_profiler/helpers/util.py:569]: @alexsherstinsky per our offline conversation, this is the only method available to us at this time, but it should be replaced once a new bootstrap algorithm implementation is completed (GREAT-705).

**Review by NathanFarmer — APPROVED** [review]
LGTM!

**Pull request merged**